src/Shared/InterningBinaryReader.cs

```diff
@@ -40,15 +40,15 @@ internal class InterningBinaryReader : BinaryReader
         /// Comment about constructing.
         /// </summary>
         private InterningBinaryReader(Stream input, Buffer buffer)
-            : base(input, buffer.Encoding)
+            : base(input, Encoding.UTF8)
```

Comment:
Why the hardcoded encoding?

Reply:
It was already hardcoded before, just in a more convoluted way, presumably due to a perf concern. I've simplified the code a tiny bit.